modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -427,11 +427,11 @@ class SeqexecEngine(httpClient: Client[IO], settings: Settings[IO], sm: SeqexecM
     } yield translator.sequence(seqId, odbSeq)
 
     def loadSequenceEvent(seqg: SequenceGen): executeEngine.EventType =
-      Event.modifyState[executeEngine.ConcreteTypes](( {st:EngineState => st.sequences.get(seqId)
-        .map(os => {
-          val newseq = os.copy(seqGen = seqg)
-          EngineState.atSequence(seqId).set(newseq) >>> updateSequenceEndo(seqId, newseq)
-        }).getOrElse(loadSequenceEndo(seqId, seqg))(st)} withEvent LoadSequence(seqId)).toHandle)
+      Event.modifyState[executeEngine.ConcreteTypes](( { st:EngineState =>
+        st.sequences.get(seqId).fold(loadSequenceEndo(seqId, seqg))(
+          _ => reloadSequenceEndo(seqId, seqg)
```

Comment:
Refresh the sequence with the new `reloadSequenceEndo`, which rebuilds the steps list, instead of using `updateSequenceEndo`, which only updates existing steps.